### ISSUE TITLE
possibly-missing-submodule false negative: function-scoped import silences warning at module level

### ISSUE THREAD

**Issue by gncs** [astral-sh/ty#3170]
﻿# Summary

An import pkg.submodule inside a function body suppresses the possibly-missing-submodule diagnostic for accesses to pkg.submodule at module level, even though the function is never called. At runtime, this is an AttributeError.

# Reproduction
```python
 # my_package/__init__.py
 # (empty)
 ```
```python
 # my_package/sub_module.py
 def f() -> None:
     print("Hello from f() in sub_module.py")
```
```python
 # ty_package.py
 import my_package
 
 
 def g() -> None:
     import my_package.sub_module
 
     print("Imported {} successfully".format(my_package.sub_module.__name__))
 
 
 my_package.sub_module.f()  # AttributeError at runtime
```
Expected: ty warns about my_package.sub_module on line 10 (module-level access, but the import only exists inside the uncalled function g()).

Actual: All checks passed!

If the line `import my_package.sub_module` is commented out, ty correctly emits possibly-missing-submodule. So the check exists but treats the import as file-wide rather than scope-aware.

# Python output:

 AttributeError: module 'my_package' has no attribute 'sub_module'

# Related issues

 - #1482 — identified the same root cause: the "first from first serve" analysis operates on the entire file rather than per-scope. Closed, but 
this case still reproduces.
 - #1488 — available_submodule_attributes priority/scoping, which was noted as a prerequisite for fixing this class of issues.

### Version

0.0.26

**Comment by carljm** [comment]
Thanks for the report!

The behavior of submodule attributes at runtime is so inconsistent and dependent on timing of which code path is executed exactly when, which isn't generally tractable for static analysis, that we aren't going to be able to achieve perfect fidelity -- there will always be examples in which we have a false positive or false negative. In particular, the question "is `g` called?" (and particularly "is it called before or after some usage of `mypackage.sub_module`?")is potentially quite difficult to answer, and I don't think is something we should attempt to track.

That said, I don't think the current behavior is great. Just because a function like this exists in the module, that doesn't imply that it is likely to be called before any given usage of `mypackage.sub_module` elsewhere in the module. I think it would be better to not consider within-function imports at all when deciding whether a submodule attribute should be considered to exist throughout the module.
